lib/html5ever/precompiled.ex

```diff
@@ -210,26 +255,48 @@ defmodule Html5ever.Precompiled do
     Enum.join(values, "-")
   end
 
+  @doc """
+  Perform the download or load of the precompiled NIF
+
+  It will look in the "priv/native/otp_app" first, and if
+  that file doesn't exist, it will try to fetch from cache.
+  In case there is no valid cached file, then it will try
+  to download the NIF from the provided base URL.
+  """
   def download_or_reuse_nif_file(rustler_opts, opts) do
     name = Keyword.fetch!(rustler_opts, :otp_app)
     version = Keyword.fetch!(opts, :version)
 
-    priv_dir = Application.app_dir(name, "priv")
+    native_dir = Application.app_dir(name, @native_dir)
 
-    cache_opts = if System.get_env("MIX_XDG"), do: %{os: :linux}, else: %{}
-    cache_dir = :filename.basedir(:user_cache, Atom.to_string(name), cache_opts)
+    cache_dir = cache_dir("precompiled_nifs")
 
     with {:ok, target} <- target() do
-      nif_name = rustler_opts[:crate] || name
-      lib_name = "#{lib_prefix(target)}#{nif_name}-v#{version}-#{target}"
+      basename = rustler_opts[:crate] || name
+      lib_name = "#{lib_prefix(target)}#{basename}-v#{version}-#{target}"
 
       file_name = lib_name_with_ext(target, lib_name)
       cached_tar_gz = Path.join(cache_dir, "#{file_name}.tar.gz")
 
-      lib_file =
-        priv_dir
-        |> Path.join("native")
-        |> Path.join(file_name)
+      lib_file = Path.join(native_dir, file_name)
+
+      base_url = Keyword.fetch!(opts, :base_url)
+      # TODO: once we move to Rustler, we probably don't need to fetch `:nif_module`
+      nif_module = Keyword.fetch!(opts, :nif_module) |> inspect()
```

Comment:
I don't think we need to inspect this. We are already evaluating the file, which means the user can inject any code they want in there, which is not different from executing code in your mix.exs anyway. So I think you have two options:

1. Keep it as an Elixir file and just eval it, with no safeguards
2. Store it in a simple CSV file

If you want to get with 2, you could name the checksum file after the module. `Elixir.Html5Ever.Native.rustler-checksum`. I will drop another comment on the path soon.

Reply:
> Keep it as an Elixir file and just eval it, with no safeguards

You mean, `Code.eval_file`? This is to simplify the loading code?

I prefer to keep as Elixir file because it's easier to work with the data here.

Reply:
I changed to use the module as atom, and also changed to eval the string directly after reading the file.